spec/friendly_uuid_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'active_record'
+require_relative '../lib/friendly_uuid'
+
+RSpec.describe FriendlyUUID do
+  before :all do
+    set_up_database
+  end
+
+  let!(:item_1) { Item.create }
+  let!(:item_2) { Item.create }
+
+  describe '#find' do
+    it 'returns an object when passed the original ID' do
+      found_item = Item.find(item_1.id)
+
+      expect(found_item).to be_kind_of Item
+    end
+
+    it 'returns an object when passed a shortened ID' do
+      found_item = Item.find(item_1.id[0])
+
+      expect(found_item).to be_kind_of Item
+    end
+
+    it 'returns an array when passed a single shortened ID as an array' do
+      found_item = Item.find([item_1.id[0]])
+
+      expect(found_item).to be_kind_of Array
+    end
+
+    it 'returns an array when passed a single ID as an array' do
+      found_item = Item.find([item_1.id])
+
+      expect(found_item).to be_kind_of Array
+    end
+
+    it 'returns an array when passed multiple shortened IDs' do
+      found_item = Item.find(item_1.id[0], item_2.id[0])
```

Comment:
We might want to use the canonical shortening code rather than `[0]`. If `item_1` and `item_2` randomly get the same first character in their IDs, then we're essentially asking `find` to find the same item twice, and it will return only one element, causing this test to fail.

```suggestion
      found_item = Item.find(item_1.to_param, item_2.to_param)
```

(Same for the other tests in this file that use `[0]`)

Reply:
Very good call. Thanks for pointing that out.

Reply:
Very good call. Thanks for pointing that out.